vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/DocViewImporter.java

```diff
@@ -989,8 +989,19 @@ private void removeReferences(@NotNull Node node) throws ReferentialIntegrityExc
         VersioningState vs = new VersioningState(stack, node);
         Node updatedNode = null;
         Optional<String> identifier = ni.getIdentifier();
+
+        boolean isUuidProtected = false;
+        if (node.hasProperty(Property.JCR_UUID)) {
+            isUuidProtected = node.getProperty(Property.JCR_UUID).getDefinition().isProtected();
+        } else if (! session.getRepository().getDescriptor(Repository.REP_NAME_DESC).contains("Oak")) {
```

Comment:
why else if? Shouldn't this also overwrite isUuidProtected from line 995 potentially?

Reply:
good question; tests passed with Jackrabbit; will have another look

Reply:
mifgt have to do with https://jackrabbit.apache.org/oak/docs/differences.html#Identifiers - but code is unclear and needs fixing

Reply:
Should better wirh https://github.com/apache/jackrabbit-filevault/pull/348/commits/223bb00051fb67b41e2a01dd2d5f70a7f5ca7a2b